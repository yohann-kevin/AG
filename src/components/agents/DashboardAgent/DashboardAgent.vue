<template>
  <div class="agent-dashboard">
    <h2>Dashboard</h2>
  </div>
</template>

<script>
export default {

}
</script>

<style>
.agent-dashboard {
  width: 100%;
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
}

.agent-dashboard {
  width: 100%;
  text-align: center;
  margin: 10px;
}
</style>
