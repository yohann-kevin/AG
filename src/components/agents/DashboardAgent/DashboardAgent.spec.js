import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuetify from "vuetify";
import DashboardAgent from "./DashboardAgent.vue";

describe("DashboardAgent", () => {
  const initWrapper = () => {
    const localVue = createLocalVue();
    const vuetify = new Vuetify();
    return shallowMount(DashboardAgent, {
      localVue,
      vuetify,
    });
  };

  it("should test component render correctly", async () => {
    const wrapper = initWrapper();
    const agents = {
      id: "f5552305-302b-4c6a-acc5-7b722225c54c",
      firstname: "Yohann",
      lastname: "Perriguey",
      email: "Yo44prg@icloud.com",
    };
    await wrapper.setData({ agents: agents });
    const title = wrapper.find("h2");
    expect(title.text()).toContain("Tableau de bord");
    expect(wrapper).toMatchSnapshot();
  });
});
