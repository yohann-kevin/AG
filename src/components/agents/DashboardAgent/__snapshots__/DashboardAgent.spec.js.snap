// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardAgent should test component render correctly 1`] = `
<div class="agent-dashboard">
  <h2>Tableau de bord</h2>
  <!---->
  <div class="agent-options">
    <v-card elevation="8" class="agent-card">
      <v-card-title>
        Vos demandes de rendez-vous
      </v-card-title>
      <v-card-text>
        Fonctionnalité indisponible pour le moment !
      </v-card-text>
    </v-card>
    <v-card elevation="8" class="agent-card">
      <v-card-title>
        Vos factures
      </v-card-title>
      <v-card-text>
        Fonctionnalité indisponible pour le moment !
      </v-card-text>
    </v-card>
    <v-card elevation="8" class="agent-card">
      <v-card-title>
        Vos informations
      </v-card-title>
      <v-card-text>
        <!---->
      </v-card-text>
    </v-card>
    <v-card elevation="8" class="agent-card">
      <v-card-title>
        Votre mot de passe
      </v-card-title>
      <v-card-text>
        <v-form>
          <v-text-field color="black" rules="v => /^(?=.*\\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/.test(v) || 'Mot de passe invalide'" type="password" label="Entrer votre ancien mot de passe" required=""></v-text-field>
          <v-text-field color="black" rules="v => /^(?=.*\\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/.test(v) || 'Mot de passe invalide'" type="password" label="Nouveau mot de passe" required=""></v-text-field>
          <v-text-field color="black" rules="true" type="password" label="Confirmer le nouveau mot de passe" required=""></v-text-field>
          <div class="update-agent-btn">
            <v-btn text="" disabled="true">
              Envoyer
            </v-btn>
            <v-btn text="" class="ml-4">
              Annuler
            </v-btn>
          </div>
        </v-form>
      </v-card-text>
    </v-card>
    <div>
      <modaldelete-stub modalinfo="[object Object]"></modaldelete-stub>
    </div>
    <v-btn dark="" text="" color="error" class="ma-2">
      Supprimer votre compte
    </v-btn>
  </div>
</div>
`;
